### ISSUE TITLE
Incorrect signature length (wanted 65, got 132)

### ISSUE THREAD

**Issue by john-osullivan** [immutability-io/vault-ethereum#5]
UPDATE: Read through the source and discovered that the `0x` prefix requirement is because of the `hexutils` implementation.  Are all transaction hashes in ethereum actually required to start `0x` when they're transmitted in hex?  It seems more like that was a safe-guard for parsing addresses that's being misapplied in this case.

I'm using this plug-in for some of proof-of-concept work around getting private keys out of `geth`/`quorum` nodes.  The goal is to have Vault generate private keys and then sign transactions. The key pieces are bridged up, but I'm not getting back signatures of the right size -- test cases are failing saying that the signature has size 132 when it needs size 65.  Size 65 also seems correct if the signature formula is two length-32 sequences followed by a recovery byte.  

This seems like it must be a configuration issue given that you're using it successfully, though.  What am I missing here?  Probably an encoding issue somewhere down the line.

## Detailed Description
I'm connecting `vault-ethereum` to a fork of `quorum` in two key places:
1. `/accounts/key#newKey()` - This is already behaving -- Vault generates the keys and, for now, hands them back for `quorum` to put into its keystore.
2. `/crypto/crypto#Sign()` is being called without issue, the data is being signed, but the resulting signatures are too large.  My replacement implementation makes a barebones call to this library's `sign` endpoint:
```golang
func (v *VaultDriver) Sign(txHash []byte) (signature []byte, err error) {
	vault := v.client.Logical()
	resp, err := vault.Write(
		v.scopedPath("sign"),
		map[string]interface{}{
			"data": txHash,
		})
	if err != nil {
		fmt.Println("err on Sign:", err.Error())
		return nil, err
	}
	signature = []byte(resp.Data["signature"].(string))
	return
}
```
The `txHash` is, as you might expect, the already-hashed transaction.  I've tried a few different versions of that sign call and run into different bugs.
1. The above call (`"data": txHash`) successfully returns a signature, but it is too large.
2. If I add `"raw": true` to the call, then the call fails with a `hex string without 0x prefix` error
3. If I then change `txHash` to `append([]string{"0x"}, string(txHash))`, then the call fails with `error converting input [0x �N�axoCA��t�hnU�J�����#�)���qg�] for field "data": '' expected type 'string', got unconvertible type '[]interface {}'`

I've been trying alternate approaches to get around the `hex string without 0x prefix` error, but nothing's working.  Would appreciate any guidance!

## Context
I'm trying to plug `vault-ethereum` into a forked version of `quorum` so that the node can sign transactions without ever actually holding the private keys itself.  If this was easy, people could easily spin up nodes without worrying about safely managing the private credentials on them.

## Environment
* Running current latest vault-ethereum version, commit 9163110
* Vault version used: v0.10.1 ('756fdc4587350daf1c65b93647b2cc31a6f119cd')
* GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
* Running on Win10 via the Windows Linux Subsystem

**Comment by john-osullivan** [comment]
Solved this issue!  Just in case somebody comes across this in the future, the key was just over twice as large because it was encoded with `go-ethereum/common/hexutils.Encode`.  Encoding doubled the size (65 => 130), then two bytes on the front for the `0x`.

The way my code was setup made it inconvenient to use the same `hexutils` library, so I just used the standard `encoding/hex` library: `signature := hex.DecodeString(res.Data["signature"].(string)[2:])`.  Worked like a charm, I'll close this out.
